src/emulator/apphosting/config.ts

```diff
@@ -11,18 +11,29 @@ import { AppHostingYamlConfig } from "../../apphosting/yaml";
  * Loads in apphosting.yaml & apphosting.local.yaml, giving
  * apphosting.local.yaml precedence if present.
  */
-export async function getLocalAppHostingConfiguration(cwd: string): Promise<AppHostingYamlConfig> {
-  const appHostingConfigPaths = discoverConfigsAtBackendRoot(cwd);
+export async function getLocalAppHostingConfiguration(
+  backendDir: string,
+): Promise<AppHostingYamlConfig> {
+  const appHostingConfigPaths = listAppHostingFilesInPath(backendDir);
   // generate a map to make it easier to interface between file name and it's path
   const fileNameToPathMap: Map<string, string> = new Map();
   for (const path of appHostingConfigPaths) {
     const fileName = basename(path);
     fileNameToPathMap.set(fileName, path);
   }
 
-  const baseFilePath = fileNameToPathMap.get(APPHOSTING_BASE_YAML_FILE)!;
+  const baseFilePath = fileNameToPathMap.get(APPHOSTING_BASE_YAML_FILE);
   const localFilePath = fileNameToPathMap.get(APPHOSTING_LOCAL_YAML_FILE);
 
-  // apphosting.local.yaml is not required to run the emulator so it may not exist
-  return await loadConfigForEnvironment(localFilePath ?? baseFilePath, baseFilePath);
+  // apphosting.local.yaml or apphosting.yaml are not required to run the emulator
+  if (!baseFilePath && !localFilePath) {
+    return AppHostingYamlConfig.empty();
+  }
+
+  // If one of them exists ...
+  if (!baseFilePath || !localFilePath) {
+    return await AppHostingYamlConfig.loadFromFile((baseFilePath || localFilePath)!);
+  }
+
+  return await loadConfigForEnvironment(localFilePath, baseFilePath);
```

Comment:
Hmm doesn't this load `apphosting.{env}.yaml`? IIUC, we don't load environment configs.

Would it make more sense to call `.merge()` here directly?